ICRCs/ICRC-103/ICRC-103.md

```diff
@@ -0,0 +1,103 @@
+| Status |
+|:------:|
+|Draft|
+
+# ICRC-191: Enhanced Allowance Query Mechanism with Pagination
+
+## 1. Introduction
+
+Although calls to the `icrc2_approve` and `icrc2_transfer_from` methods are recorded in the ledger, it is not possible to determine the allowances that are in effect at some point in time, except by traversing the entire ledger.  This standard introduces an endpoint which will return this information thus making the management of allowances feasible.
+
+ICRC-191 is an extension of the ICRC-2 standard.  
+ICRC-191 specifies a way to list outstanding allowances.
+
+
+## 2. Metadata
+
+A ledger that implements ICRC-191 MUST include `record {name = "ICRC-191"; url = "https://github.com/dfinity/ICRC-1/standards/ICRC-191"}` as part of the output of `icrc1_supported_standards`.
+
+The endpoint introduced in this standard operates in two ways.  In the public version any principal can obtain the outstanding allowances of any other principal. In the private version, the allowances returned by the endpoint must have been issued by the caller (i.e. the caller is the principal controlling the source account of an allowance.)
+Which version of the standard is implemented by a ledger is specified through metadata which can be retrieved using `icrc1_metadata`.
+
+A ledger that implements ICRC-191 MUST return metadata `icrc191:public_allowances` of type Text (optional). The possible values are "true" if the allowance data is public and "false" otherwise.
```

Comment:
Good point -- there is not good reason why we should make the value optional. I'll remove that

Reply:
> Good point -- there is not good reason why we should make the value optional. I'll remove that

On another note, in ICRC-7 we also had a query method to get the same value directly without having to fetch and parse the whole collection metadata.

Reply:
Is this icrc7_collection_metadata?  Why "collection"? :)

Reply:
We decided at the time to clarify where each metadata belongs to by calling it either collection or token metadata but never metadata without collection or token. We could also have opted for metadata and token metadata, but we were concerned that this would make the documentation hard to follow.

Reply:
sounds good -- I've added the method per your suggestion